extractor/src/main/java/org/schabi/newpipe/extractor/linkhandler/LinkHandlerFactory.java

```diff
@@ -42,12 +42,30 @@ public String getUrl(String id, String baseUrl) throws ParsingException {
     // Logic
     ///////////////////////////////////
 
+    /**
+     * Builds a {@link LinkHandler} from a url.<br>
+     * Be sure to call {@link Utils#followGoogleRedirectIfNeeded(String)} on the url if overriding
+     * this function.
+     * @param url the url to extract path and id from
+     * @return a {@link LinkHandler} complete with information
+     */
     public LinkHandler fromUrl(String url) throws ParsingException {
         if (url == null) throw new IllegalArgumentException("url can not be null");
+        url = Utils.followGoogleRedirectIfNeeded(url);
```

Comment:
The input parameter shouldn't be reassigned. Especially since with the new checkstyle rules in the NewPipe repo forcing everything input parameter to be final.